src/typechecker.erl

```diff
@@ -2823,6 +2823,10 @@ type_check_assocs(Env, [{Assoc, P, Key, Val}| Assocs])
 type_check_assocs(_Env, []) ->
     {[], #{}, constraints:empty()}.
 
+update_map_type({ann_type, _, [_, Ty]}, AssocTys) ->
+    %% FIXME we really should get rid of `ann_type's right in the
+    %% begining or during normalization.
```

Comment:
Better not add this, considering your other PR. :stuck_out_tongue_winking_eye:

Reply:
I have low hopes of addressing the other PR soon, so I thought until then let's add a quick-fix.

Reply:
Right now we ignore `ann_type`s and I think that's the wrong thing to do. We really should check them when they show up, otherwise people are going to be very confused.
But I'm not saying that you have to change that for this PR, but I don't think removing the `ann_type`s is the right way forward.

Reply:
let's discuss the role of annotated types in Gradualizer in #180